int8 isActive
float64 data 
